### PR TITLE
Bugfix/GitHub 1172 suite sparse

### PR DESCRIPTION
**Pull request by lee218llnl**
Proposed fix for #1172

### CODE REVIEW

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 1]
```diff
@@ -41,33 +42,35 @@ class SuiteSparse(Package):
     depends_on('lapack')
 
     depends_on('metis@5.1.0', when='@4.5.1')
+    depends_on('metis@5.1.0', when='@4.5.3')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: i believe you can just modify the previous line to be `when='@4.5.1:'`

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 2]
```diff
@@ -78,10 +81,13 @@ def install(self, spec, prefix):
             ])
 
         # BLAS arguments require path to libraries
-        # FIXME : (blas / lapack always provide libblas and liblapack as aliases)
-        make_args.extend([
-            'BLAS=-lblas',
-            'LAPACK=-llapack'
-        ])
+        # FIXME: (blas/lapack always provide libblas and liblapack as aliases)
+        if '@4.5.1' in spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: you sure you don't need to specify `BLAS=-lblas` and `LAPACK=-llapack` for `4.5.3`?
Since you modify these lines, would you mind doing
```
to_link_flags(spec['lapack'].lapack_shared_lib)
to_link_flags(spec['blas'].blas_shared_lib)
```
instead (i.e. avoid hardcoding `lblas` and `llapack`)?

**Comment by davydden** [comment]
i would squash the three last commits as there is no reason for them to be independent.

File: var/spack/repos/builtin/packages/suite-sparse/package.py (hunk 1 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: yup, will do

File: var/spack/repos/builtin/packages/suite-sparse/package.py (hunk 2 above)
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: the make_args.extend was in the original suite-sparse package for version 4.5.1, so I'm not sure if we want to change that and risk breaking previous users. I'm also not confident that the suite-sparse build system will obey the to_link_flags. The 4.5.3 version does not require these flags (I think they fixed their makefile).

Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: essentially `BLAS=-lblas` and `LAPACK=-llapack` are link flags, as far as i can tell. So there should be no harm using something like `BLAS=-L/path/to/spack/opt/openblas/lib -lopenblas` and alike (exactly what `to_link_flags(spec['blas'].blas_shared_lib)` will resolve to). Moreover, this is more correct as otherwise suite-sparse may link against some other blas/lapack (e.g. system-provided).
But this can be left for another PR.

W.r.t. 4.5.3., can you please check if resulting libraries are linked against Spack's blas/lapack without those config options?

**Comment by tgamblin** [comment]
Is this ready to merge after the squash?

**Comment by lee218llnl** [comment]
OK, I think it's good to go from my perspective. I didn't change the to_link_flags, though, so I hope @davydden is OK with that.

**Comment by davydden** [comment]
> so I hope @davydden is OK with that.

I am fine with that, i will test it later.

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 3]
```diff
@@ -33,41 +33,43 @@ class SuiteSparse(Package):
...
+    version('4.5.3', '8ec57324585df3c6483ad7f556afccbd')
 
     # FIXME: (see below)
     # variant('tbb', default=True, description='Build with Intel TBB')
 
     depends_on('blas')
     depends_on('lapack')
 
-    depends_on('metis@5.1.0', when='@4.5.1')
+    depends_on('metis@5.1.0', when='@4.5.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: Are pre-4.5.1 versions not dependent on metis? Why not always depend on metis?

Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: i think they have somewhere an officially recommended metis for each suite-sparse release.

Comment (tgamblin) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: I think this can be a separate PR
